compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -538,9 +514,31 @@ fn encode_ty<'tcx>(
         // User-defined types
         ty::Adt(adt_def, substs) => {
```

Comment:
I believe this misses `#[repr(transparent)]` support.

Reply:
Is it in `transform_ty`? How exactly does that function relate to `encode_ty` and why is it split?

Reply:
Yes, it's handled in `transform_ty`, and its tests are here: https://github.com/rust-lang/rust/blob/master/src/test/codegen/sanitizer-cfi-emit-type-metadata-id-itanium-cxx-abi.rs#L116-L132. `transform_ty` coalesces types that have the same encoding into a single entity (e.g., c_void into unit), normalizes, and generalizes types before handing them to `encode_ty`. It greatly simplifies the encoding and compression/substitution logic.